python/packages/nisar/products/insar/utils.py

```diff
@@ -370,6 +372,128 @@ def _compute_subswath_mask_id(azi_idx,
+    hdf5_ds_obj : h5py.Dataset
+        The HDF5 dataset object
+    lines_per_block : integer (default: 1000)
+         Lines per block to write the data to the hard drive
+    """
+
+    input_src = gdal.Open(input_file_path)
+    width = input_src.RasterXSize
+    length = input_src.RasterYSize
+
+    # Write data block by block
+    for line in range(0, length, lines_per_block):
+        line_blocks = lines_per_block
+        if (line + lines_per_block) > length:
+            line_blocks = length - line
+        data = input_src.GetRasterBand(1).ReadAsArray(0,line, width, line_blocks)
+        hdf5_ds_obj.write_direct(data,
+                                 dest_sel=np.s_[line : line + line_blocks, : width])
+
+    input_src = None
+
+def generate_dem_rdr(radar_grid_obj,
+                     orbit_obj,
+                     dem_file,
+                     out_dem_rdr_path,
+                     use_gpu = True,
+                     dem_interp_method = 'BIQUINTIC',
+                     threshold = 1.0e-7,
+                     numiter = 25,
+                     extraiter = 10,
+                     lines_per_block = 1000):
+    """
+    Generate the DEM in radar grid
+
+    Parameters
+    ---------
+    radar_grid_obj : SLC
+        The radar grid object for the reference RSLC
+    orbit_obj : isce3.core.Orbit
+        The SLC object for the secondary RSLC
+    dem_file  : str
+        DEM file
+    out_dem_rdr_path : str
+        output path of the DEM in radar grid
+    use_gpu : boolean (default: True)
+        Indicator to use the GPU or not
+    dem_interp_method : str (default: BIQUINTIC)
+        DEM interpolation method, one of 'BILINEAR', 'BICUBIC', 'NEAREST', and 'BIQUINTIC'
+    threshold : float (default: 1.0e-7)
+        The rdr2geo absolute slant range convergence tolerance (m)
+    numiter : integer (default: 25)
+        Maximum number of primary Newton-Raphson iterations
+    extraiter : integer (default: 10)
+         Maximum number of secondary iterations
+    lines_per_block : integer (default: 1000)
+         Lines per block to write the generated DEM to the hard drive
+    """
+
+    error_journal = journal.error('utils.generate_insar_dem')
+    grid_doppler = isce3.core.LUT2d()
+
+    dem_raster = isce3.io.Raster(dem_file)
+    if dem_raster is None:
+        err_str = f'Can not open the DEM file {dem_raster}'
+        error_journal.log(err_str)
+        raise ValueError(err_str)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+
+    # Default DEM interpolation method is BIQUINTIC
+    interp_method = isce3.core.DataInterpMethod.BIQUINTIC
+    if dem_interp_method == 'BILINEAR':
+        interp_method = isce3.core.DataInterpMethod.BILINEAR
+    if dem_interp_method == 'BICUBIC':
+        interp_method = isce3.core.DataInterpMethod.BICUBIC
+    if dem_interp_method == 'NEAREST':
+        interp_method = isce3.core.DataInterpMethod.NEAREST
```

Comment:
Looks like if the user of this function passes anything beyond "BILINEAR", "BICUBIC" and "NEAREST", the interp_method is set to BIQUINTIC. Even if user passes dem_interp_method = 'ANYTHING unrelated', it will still silently ignored and set to BIQUINTIC. I don't think if that is a correct begavior. We should let the underlaying functions throw an exception if the interolation method is not valid. I suggest the following minor change:

```suggestion
    
   try:
         interp_method = getattr(isce3.core.DataInterpMethod, dem_interp_method)
    except AttributeError:
         raise ValueError(f"invalid interpolation method: {dem_interp_method}")
```